### PR TITLE
[Program: GCI] Cron job to remove unverified users after a time period

### LINKED ISSUE #115: Remove unverified users after some time has passed since registration
## Description
As a developer,
I need avoid test or invalid users on the database,
so that I can save up storage space.

We need to develop a scheduler/ cron job to delete user with unverified email for some period of time.
**The period still needs to be discussed** with the community and admins.

### Update [Required]
- [ ] Cron job to delete unverified users after a certain period of time

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
3 hours

### PR DESCRIPTION
**Pull request by daksha-aeer**
### Description
Cron job created to remove unverified users after a time period. Code is in **"remove_unverified_users_cron_job.py"** and scheduler in **"background_scheduler.py"**. 

#### Working:
- unverified_users queried from database with UserModel
- Difference between current and registration time found. If it is greater than VERIFICATION_TIME, delete user.

Fixes #115 

### Type of Change:
- Code

**Code/Quality Assurance Only**
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
1. New test case added in **test_remove_unverified_users_cron.py**

2. Using print statement    
print() statements added to display unverified users. VERIFICATION_TIME set to 20 days. 

*Print statements are not in PR*
```py
        for user in unverified_users:
            register_time = datetime.fromtimestamp(user.registration_date)
            time_difference = current_time - register_time
            print("%s : pending for %s days"%(user.name,time_difference.days))
            if time_difference > VERIFICATION_TIME:
                print("removing %s--------------------------"%user.name)
                UserModel.delete_from_db(user)
        print("--------------------done-----------------------")
```
1. Before
task45_1

2. After
task45_2

3. Tests
```sh
python -m unittest discover tests
```
task45_3

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Postman API at /docs folder
- [ ] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: app/schedulers/remove_unverified_users_cron_job.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    """
+    This function iterates of all the unverified users and
+    checks if email is not verified beyond a period of time.
+    if True then user is removed, if False does nothing
+    """
+    VERIFICATION_TIME = timedelta(days = 20)
+    from run import application
+    with application.app_context():
+        from app.database.models.user import UserModel
+        unverified_users = UserModel.query.filter_by(is_email_verified = False)
```
Comment (isabelcosta) [inline app/schedulers/remove_unverified_users_cron_job.py:14]: Great idea! Querying by email verified before performing the loop here in python code 👏

**Review by isabelcosta — APPROVED** [review]
Great job @daksha-aeer !
Your "How has this been tested?" section is really well detailed 👏 
Thank you so much for your contribution!

File: app/schedulers/background_scheduler.py [hunk 2]
```diff
@@ -11,9 +11,17 @@ def init_scheduler():
...
+    scheduler.add_job(id='remove_unverified_cron', func=complete_remove_unverified_users_job,
+                      trigger='cron', hour=23, minute=59, second=0, day='*', timezone='Etc/UTC',
+                      replace_existing=True)
+
     # for tests purposes
     # scheduler.add_job(id='complete_mentorship_relations_cron', func=complete_overdue_mentorship_relations_job,
     #                   trigger='interval', seconds=4,
     #                   replace_existing=True)
 
+    # scheduler.add_job(id='remove_unverified_cron', func=complete_remove_unverified_users_job,
```
Comment (annabauza) [inline app/schedulers/background_scheduler.py:23]: do we need this comment?

**Review by annabauza — CHANGES_REQUESTED** [review]
Please add some tests

File: app/schedulers/background_scheduler.py (hunk 2 above)
Comment (daksha-aeer) [reply app/schedulers/background_scheduler.py:23]: Hi Anna! This part is for test purposes, similar to the one above. I've added the explanation in the PR

Comment (daksha-aeer) [reply app/schedulers/background_scheduler.py:23]: The cron job runs every 24 hours. But for testing purposes the time period is 4 seconds. This is the commented code.

**Comment by daksha-aeer** [comment]
> Please add some tests

I've added test cases. Please check new file **test_remove_unverified_users_cron.py**

**Comment by annabauza** [comment]
closing due to merge #350

**Pull request closed without merging**